### PR TITLE
Fix chat editor overflow

### LINKED ISSUE #234902: Paste options is behind attach btn
* use edits
* paste a function (see below) into the input
* :bug: the paste options are rendered behind the attach btn

Image

```
export function fib(nth) {
	if (nth <= 0) return 0;
	if (nth === 1) return 0;
	if (nth === 2) return 1;

	return fib(nth - 1) + fib(nth - 2);
}
```

### PR DESCRIPTION
**Pull request by mjbvz**
Use layout service instead

Fixes #234902

### CODE REVIEW

**Review by Tyriar — DISMISSED** [review]

File: src/vs/workbench/contrib/chat/browser/chatViewPane.ts [hunk 1]
```diff
@@ -155,7 +157,9 @@ export class ChatViewPane extends ViewPane implements IViewWelcomeDelegate {
 
 			const scopedInstantiationService = this._register(this.instantiationService.createChild(new ServiceCollection([IContextKeyService, this.scopedContextKeyService])));
 			const locationBasedColors = this.getLocationBasedColors();
-			const editorOverflowNode = $('.chat-editor-overflow-widgets');
+			const editorOverflowNode = this.layoutService.mainContainer.appendChild($('.chat-editor-overflow.monaco-editor'));
```
Comment (bpasero) [inline src/vs/workbench/contrib/chat/browser/chatViewPane.ts:160]: ✋ how does this work with floating windows? `mainContainer` is only main window 🤔

**Review by bpasero — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/chat/browser/chatViewPane.ts (hunk 1 above)
Comment (mjbvz) [reply src/vs/workbench/contrib/chat/browser/chatViewPane.ts:160]: Good point. In this case though I think the chat view pane can only exist in the main window, right?

Comment (bpasero) [reply src/vs/workbench/contrib/chat/browser/chatViewPane.ts:160]: Chat can be opened as editor, and thus move into floating windows.

**Comment by justschen** [comment]
looks like this works bettter >>> hard coding the z-index. rip the CI though, not sure what's going on there :(

**Review by justschen — APPROVED** [review]

**Review by jrieken — APPROVED** [review]

**Review by bpasero — APPROVED** [review]

**Pull request merged**